src/core/tsi/ssl_transport_security_utils.h

```diff
@@ -0,0 +1,147 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_TSI_SSL_TRANSPORT_SECURITY_UTILS_H
+#define GRPC_CORE_TSI_SSL_TRANSPORT_SECURITY_UTILS_H
+
+#include <grpc/support/port_platform.h>
+
+#include <openssl/x509.h>
+
+#include "absl/strings/string_view.h"
+
+#include <grpc/grpc_security_constants.h>
+
+#include "src/core/tsi/ssl/key_logging/ssl_key_logging.h"
+#include "src/core/tsi/transport_security_interface.h"
+
+namespace grpc_core {
+
+// Converts an SSL error status code to a readable string.
+//
+// error: the SSL error status code.
+//
+// return: the corresponding status string.
+const char* SslErrorString(int error);
+
+// Logs the SSL error stack.
+void LogSslErrorStack(void);
+
+// Performs an SSL_write and handle errors.
+//
+// ssl: the SSL object to write to.
+// unprotected_bytes: the buffer containing the bytes for writing to |ssl|.
+// unprotected_bytes_size: the size of the buffer |unprotected_bytes|.
+//
+// return: TSI_OK if the write operation succeeds or corresponding TSI errors.
+tsi_result DoSslWrite(SSL* ssl, unsigned char* unprotected_bytes,
```

Comment:
In the long run, I think we will want to eliminate `tsi_result` in favor of `absl::Status`.  Since this PR is converting these functions to idiomatic C++, would it make sense to do that here?  Or would it make more sense to keep using `tsi_result` for now, and then in a future PR convert all uses for `tsi_result` to `absl::Status` all at once?

Reply:
I prefer to make those changes in a separate PR. I looked up the list of `tsi_result` values and `absl::Status` and I didn't see a perfect match between these two lists. I think more work is needed to do this conversion so it is better to leave them to future PRs.